File: 1-two-sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    map = {}
    nums.each_with_index do |val,idx|
        if map.key?(target - nums[idx])
            return [map[target - nums[idx]], idx]
        end
        map[nums[idx]] = idx
    end
    return []
end

File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let mp = new Map();
    
    for(let i = 0; i < nums.length; i ++) {
        if(mp.has(target - nums[i])) {
            return [mp.get(target - nums[i]), i];
        }
        mp.set(nums[i], i);
    }
	return [];
};